subsys/dfu/boot/Makefile

```diff
@@ -0,0 +1,2 @@
+obj-$(CONFIG_MCUBOOT_IMG_MANAGER) += mcuboot.o
```

Comment:
You are using CONFIG_MCUBOOT_IMG_MANAGER before it was defined in Kconfig.  Please define Kconfig variable in the same commit they are being used.